File: Java Hangman/Hangman.java
import java.io.Console;
import java.util.Arrays;

public class Hangman {

  public static void main(String[] args) {
    Console console = System.console();

    console.printf("WELCOME TO HANGMAN\n");
    String word = console.readLine("List a word: ");
    String guessed_letters = "";
    String guessed_word = "";
    int guesses_left = 7;
    String guess;
    String[] letters = {"a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q",
    "r", "s", "t", "u", "v", "w", "x", "y", "z"};

    for (char letter: word.toCharArray()) {
      guessed_word += " - ";
    }
    console.printf("\nGuessed word so far: %s\n", guessed_word);

    while (guesses_left > 0) {
      if (word.equals(guessed_word)) {
        console.printf("Nice job! You guessed the word with %s guesses remaining!\n", guesses_left);
        break;
      } else {
        guessed_word = "";
        console.printf("Guesses left: %s\n",guesses_left);
        guess = console.readLine("Guess a letter: ");
        while (guess.length() > 1 || Arrays.asList(letters).indexOf(guess) == -1) {
          guess = console.readLine("Guesses must be letter and must be one character long, guess again: ");
        }
        if (word.indexOf(guess) == -1) {
          console.printf("Sorry, \'%s' is not in the word. You lost a guess!\n\n", guess);
          guesses_left -=1;
        } else {
          console.printf("Good job! \'%s' is in the word.\n\n", guess);
        }
        guessed_letters += guess;
        for (char letter : word.toCharArray()) {
          if (guessed_letters.indexOf(letter) > -1) {
            guessed_word += letter;
          } else {
            guessed_word += " - ";
          }
        }
        console.printf("Guessed word so far: %s\n",guessed_word);
    }
  }
  if (guesses_left == 0) {
    console.printf("Sorry, you lost. The word was \'%s\'. Better luck next time!\n",word);
  }
}
}

File: Python Hangman/hangman.py
import os
import random
file = open('words.txt', 'r')
words = []
for line in file.read().split('\n'):
    if len(line) >= 5:
        words.append(line.strip())

#Takes guessing word and clears screen to hide it
print("WELCOME TO HANGMAN")
word = random.choice(words)

#Game variables
correct_guessed_letters = []
wrong_guessed_letters =  []
guessed_word = ""
guesses_left = ['hat', 'head', 'torso', 'first arm', 'second arm', 'first leg', 'second leg']
letters = 'abcdefghijklmnopqrstuvwxyz'
welcome = "Guessed word so far: "

#Creates welcome message with some parameters
print("WELCOME TO HANGMAN\n")
for letter in word:
    if letter in letters:
        welcome += "-"
    elif letter not in letters:
        welcome += letter
print(welcome)

#Loop to take guess and evaluate it
while len(guesses_left) > 0:
    print("Correct letters: {}".format(', '.join(correct_guessed_letters)))
    print("Incorrect letters: {}".format(', '.join(wrong_guessed_letters)))
    print("Guesses left: {}".format(len(guesses_left)))
    guess = input("Guess a letter: ")

    #While guess has already been gueessed
    while guess in correct_guessed_letters or guess in wrong_guessed_letters:
        guess = input("You already guessed this letter, try again: ")
    #While guess is not valid
    while len(guess) != 1 or guess not in letters:
        guess = input("Guesses must be letters and must be one character long, try again: ")
    while guess in correct_guessed_letters or guess in wrong_guessed_letters:
        guess = input("You already guessed this letter, try again: ")

    #Evaluates if guess is correct or not
    if guess in word:
        print("Nice job! '{}' is in the word.\n".format(guess))
        correct_guessed_letters.append(guess)
    else:
        print("Too bad! '{}' is not in the word. You lost your {}\n".format(guess, guesses_left[0]))
        wrong_guessed_letters.append(guess)
        guesses_left.pop(0)
    guessed_word = ""

    #Updates the guessed word variable and prints it out
    for letter in word:
        if letter in correct_guessed_letters or letter not in letters:
            guessed_word += letter
        elif letter not in correct_guessed_letters and letter in letters:
            guessed_word += "-"
    print("Guessed word so far: {}".format(guessed_word))

    #Win condition
    if guessed_word == word:
        print("NICE JOB! YOU WON THE GAME WITH {} GUESSES REMAINING!!!!!!!".format(len(guesses_left)))
        break

#Lose condition
if len(guesses_left) == 0:
    print("Sorry, you lost. The word was '{}'. Better luck next time.".format(word))
